server/hotrod/src/main/scala/org/infinispan/server/hotrod/ClientListenerRegistry.scala

```diff
@@ -170,6 +197,19 @@ class ClientListenerRegistry(configuration: HotRodServerConfiguration) extends L
       cacheEventConverterFactories.clear()
    }
 
+   def findAndWriteEvents(channel: Channel): Unit = {
+      val eventSender = eventSenders.values().filter {
+         case s: BaseClientEventSender => s.hasChannel(channel)
+         case _ => false
+      }
+      eventSender.headOption.map {
+         case s: BaseClientEventSender => s.writeEventsIfPossible()
+         case _ => false
+      }
+   }
```

Comment:
Is it possible that more than one event sender is associated with the channel? 

Another way of writing it is replacing the two pattern matches with a single collect:

```scala
eventSenders.values().collectFirst { case s: BaseClientEventSender =>
         if(s.hasChannel(channel)) s.writeEventsIfPossible()
}
```
or using ```collect``` instead of ```collectFirst``` to assume that a channel can be part of more than one eventListener